tests/remote_method_test.cpp

```diff
@@ -312,7 +320,76 @@ BOOST_AUTO_TEST_CASE(testRemoteOperationCallerFactory)
 
     // this line may not crash:
     mvoid();
+}
+
+BOOST_AUTO_TEST_CASE(testRemoteOperationCallerSleep)
+{
+    OperationCaller<int(int)> sleepAndIncOwnThread("sleepAndIncOwnThread", caller->engine());
+
+    boost::shared_ptr<DisposableInterface> implementation( new RemoteOperationCaller<int(int)>(tc->provides("methods")->getPart("sleepAndIncOwnThread"), "sleepAndIncOwnThread", 0) );
+    sleepAndIncOwnThread = implementation;
+    BOOST_CHECK( sleepAndIncOwnThread.ready() );
+
+    i = 0;
+    sleepAndIncOwnThread.call(1);
+    sleepAndIncOwnThread.call(1);
+    sleepAndIncOwnThread.call(1);
+    BOOST_CHECK_EQUAL( 3, i );
+
+    i = 0;
+    sleepAndIncOwnThread.send(1);
+    sleepAndIncOwnThread.send(1);
+    sleepAndIncOwnThread.send(1);
+    wait_for_equal(3, i, 50 /* * 100ms */ );
+}
 
+BOOST_AUTO_TEST_CASE(testRemoteOperationCallerReferences)
+{
+    OperationCaller<const void *(const int&)> returnAddressOfConst("returnAddressOfConst", caller->engine());
+    OperationCaller<void *(int&)> returnAddressOf("returnAddressOf", caller->engine());
+
+    boost::shared_ptr<DisposableInterface> implementation( new RemoteOperationCaller<const void *(const int&)>(tc->provides("methods")->getPart("returnAddressOfConst"), "returnAddressOfConst", 0) );
+    returnAddressOfConst = implementation;
+    BOOST_CHECK( returnAddressOfConst.ready() );
+
+    implementation.reset( new RemoteOperationCaller<void *(int&)>(tc->provides("methods")->getPart("returnAddressOf"), "returnAddressOf", 0) );
+    returnAddressOf = implementation;
+    BOOST_CHECK( returnAddressOf.ready() );
+
+    int var = 0;
+
+    // call
+    BOOST_CHECK_EQUAL(&var, returnAddressOf(var));
+    BOOST_CHECK_EQUAL(&var, returnAddressOfConst(var));
+
+    // send
+    // Collecting from a RemoteOperationCaller returning void * segfaults!
```

Comment:
This might be another bug and I have not found the root cause yet. There is a comment about RemoteOperationCaller not supporting reference return types in [rtt/internal/DataSourceStorage.hpp:133](https://github.com/orocos-toolchain/rtt/blob/master/rtt/internal/DataSourceStorage.hpp#L133), so maybe this also affects pointer return types.